### DISCUSSION TITLE
What's the migration path for EventRegistrationToken on .Net 6 + WinUI?

### DISCUSSION THREAD

**Discussion by jeanplevesque** [unoplatform/uno#10975] [Q&A] [answered]
While migrating from Xamarin to .Net6.0, it seems that on `net6.0-android`, both `EventRegistrationToken` and `EventRegistrationTokenTable<>` are not available (using the `System.Runtime.InteropServices.WindowsRuntime` nuget package).
What would be the migration path?
The strange part is that this problem only affects Android (iOS seems fine).

![image](https://user-images.githubusercontent.com/39710855/212339949-641d5eb5-f0db-41a6-83fd-2e89f5e88202.png)

**Comment by jeromelaban** [comment, marked as answer]
This particular class was part of mscorlib in Xamarin, making the `System.Runtime.InteropServices.WindowsRuntime` unused, but starting from .NET 6, net6.0-ios stops allowing for xamarinios targets to be used, but net6.0-android does not.

The result is that considering this:
<img width="265" alt="image" src="https://user-images.githubusercontent.com/5839577/212411928-d17f3374-08e7-4005-ac19-030993bd1a19.png">

iOS falls back on the netstandard1.3 binary, but the Android one goes to an empty folder, causing the type to not be found.

The question is more about why this type is used (Uno does not use it on any target) and if it could be replaced by a local internal copy (if it's not exposed to consumers of your library).

**Reply by jeanplevesque** [reply]
It's actually used in 2 classes from a package that we're currently trying to migrate to .Net 6. Funny enough, the classes were actually written by @dr1rrb. Perhaps he might be able to guide us towards a possible refactoring or workaround?
They're called `CollectionChangedFacet` and `SelectionFacet`.

**Reply by dr1rrb** [reply]
I removed them on not supported platform : https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Facets/CollectionChangedFacet.cs#L48 where `USE_EVENT_TOKEN` is defined only for Windows UWP (not even WinUI) : https://github.com/unoplatform/uno.extensions/blob/ba5dbb7377b175801786ed47461425c3a8f15720/src/Uno.Extensions.Reactive.UI/common.props#L8

**Reply by jeanplevesque** [reply]
So I imported the version that is internal in Uno.Extensions to limit our refactoring effort. https://github.com/unoplatform/uno.extensions/blob/8ff7944257b70cd06050147ed66348d2cb996f13/src/Uno.Extensions.Reactive.UI/_Compat/EventRegistrationToken.cs

Thanks!
